### DISCUSSION TITLE
next-build: don't generate `next-env.d.ts` if if already exists

### DISCUSSION THREAD

**Discussion by juliusmarminge** [vercel/next.js#38572] [Ideas]
### Describe the feature you'd like to request

At build, Next.js automatically generates the `next-env.d.ts` in project root, see demo here: https://share.cleanshot.com/nLYI2T

The "problem" is, I place my other types in for example `src/types` (e.g. when augmenting Next-Auth.js module to include more stuff in the user as discussed in [t3-oss/create-t3-app#180](https://github.com/t3-oss/create-t3-app/pull/180).

I would like to keep them in the same place. Currently, the `next-env.d.ts` has to live in one place (project root) where as my other types lives elsewhere (src/types). 

### Describe the solution you'd like

If found somewhere in the project, the `next-env.d.ts` shouldn't have to be generated at build but instead use the available one. This could either be by explicitly telling Next where to find the types or internally looked up at build time before creating a "copy".

Pseudo example of what I mean:
```ts
const generateTypeDef = () => {
  if (findInRepo("next-env.d.ts") {
    return;
  }
  else {
    fs.copy("next-env.d.ts", projectRoot); // or generate it in some way?
  }
}
```

For example, in the demo linked above, the types exists so they won't have to be generated at build time

### Describe alternatives you've considered

I could keep it like it is today:

```
# root
src
  types
    next-auth.d.ts
  ...
next-env.d.ts
...
```

or even keep everything in root
```
# root
src
  ...
next-auth.d.ts
next-env.d.ts
...
```

but this is messy imo. I would rather gather all of them elsewhere (especially if having lots of types)

**Comment by bhatvikrant** [comment]
Keeping all the types in a dedicated folder is definitely cleaner, especially as the codebase grows there will be many `.d.ts` files.

**Comment by juliusmarminge** [comment]
Any updates on this?
